mso/resource_mso_tenant.go

```diff
@@ -355,7 +389,15 @@ func resourceMSOTenantCreate(d *schema.ResourceData, m interface{}) error {
 				mapSite["siteId"] = fmt.Sprintf("%v", inner["site_id"])
 			}
 			if inner["vendor"] != "" {
-				if inner["vendor"] == "aws" {
+				if inner["vendor"] == "gcp" {
+
+					if inner["gcp_project_id"] == "" {
+						return fmt.Errorf("gcp_project_id is required with vendor = gcp")
+					}
+
+					setGcpAccountDetails(mapSite, inner, tenantAttr.Name, true)
+
+				} else if inner["vendor"] == "aws" {
```

Comment:
if this part of the code is exactly same and repeated more than once is it possible to put it in a function?